### PR TITLE
Fixes Hub UI with latest version of `react-router-dom`

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [ ] Run API Unit Tests, Lint Checks, API Design, Golden Files with `make api-check`
- [ ] Run UI Unit Tests, Lint Checks with `make ui-check`
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/main/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: ui/src/containers/BasicDetails/index.tsx [hunk 1]
```diff
@@ -40,14 +40,9 @@ import './BasicDetails.css';
 
 const BasicDetails: React.FC = () => {
   const { resources, catalogs } = useMst();
-  const {
-    catalog,
-    kind,
-    name,
-    version
-  }: { catalog: string; kind: string; name: string; version: string } = useParams();
+  const { catalog, kind, name, version } = useParams();
```
Comment (vinamra28) [inline ui/src/containers/BasicDetails/index.tsx:43]: why is this changed?

File: ui/src/containers/Details/index.tsx [hunk 2]
```diff
@@ -13,21 +13,16 @@ import { IVersion } from '../../store/resource';
...
-    catalog,
-    kind,
-    version
-  }: { name: string; catalog: string; kind: string; version: string } = useParams();
+  const { name, catalog, kind, version } = useParams();
 
   React.useEffect(() => {
     if (resources.isResourceLoading === false) {
-      const resourceKey = `${catalog}/${titleCase(kind)}/${name}`;
+      const resourceKey = `${catalog}/${titleCase(kind as string)}/${name}`;
```
Comment (vinamra28) [inline ui/src/containers/Details/index.tsx:20]: same as above

File: ui/src/containers/ParseUrl/index.tsx [hunk 3]
```diff
@@ -20,7 +20,7 @@ const ParseUrl: React.FC = () => {
       };
       user.authenticate(codeFinal);
       if (user.isAuthenticated) {
-        history.goBack();
+        history('-1');
```
Comment (vinamra28) [inline ui/src/containers/ParseUrl/index.tsx:23]: can we add a comment here so that anyone looking at code can get to know what `-1` signifies

File: ui/src/containers/Rating/index.tsx [hunk 4]
```diff
@@ -12,8 +12,8 @@ import { titleCase } from '../../common/titlecase';
 import './Rating.css';
 
 const Rating: React.FC = observer(() => {
-  const { name, kind, catalog }: { name: string; kind: string; catalog: string } = useParams();
-  const resourceKey = `${catalog}/${titleCase(kind)}/${name}`;
+  const { name, kind, catalog } = useParams();
+  const resourceKey = `${catalog}/${titleCase(kind as string)}/${name}`;
```
Comment (vinamra28) [inline ui/src/containers/Rating/index.tsx:16]: same as above

File: ui/package.json [hunk 5]
```diff
@@ -27,7 +27,7 @@
     "react-dom": "^16.14.0",
     "react-hotkeys-hook": "^3.4.7",
     "react-markdown": "^5.0.3",
-    "react-router-dom": "^5.2.0",
+    "react-router-dom": "^6.8.0",
```
[the remark this answers is not in the document]
Comment (PuneetPunamiya) [reply ui/package.json:30]: This is bumped by the dependent bot https://github.com/tektoncd/hub/pull/782/commits/bd3d8abbd05aff55af2a270dae5e33d404033b55

File: ui/src/containers/BasicDetails/index.tsx (hunk 1 above)
Comment (PuneetPunamiya) [reply ui/src/containers/BasicDetails/index.tsx:43]: With react router dom update, this was giving a type error, hence added a change
